frontend/src/components/Schedules/EditScheduleForm.tsx

```diff
@@ -54,65 +61,96 @@ export const EditScheduleForm: FC<Props> = ({ onUpdate, schedule }) => {
-            onUpdate(ScheduleKey.INTERVAL, selected.value);
-          }}
-        />
-      </div>
+          <Input
+            onChange={({ currentTarget: { value } }) => onUpdate(ScheduleKey.DESCRIPTION, value)}
+            name={intl.get('description')}
+            defaultValue={schedule.description}
+            placeholder={intl.get('description')}
+            css=""
+          />
+        </div>
+
+        <div className={flexWrapping}>
+          <InlineFormLabel tooltip={intl.get('report_time_description')}>{intl.get('report_time')}</InlineFormLabel>
+
+          <TimeOfDayPicker
+            onChange={(selected: DateTime) => {
+              onUpdate(ScheduleKey.TIME_OF_DAY, selected.format('HH:mm'));
+            }}
+            value={formatTimeToDate(schedule?.time)}
+          />
+        </div>
 
-      <div className={flexWrapping}>
-        <InlineFormLabel tooltip={intl.get('report_group_description')}>{intl.get('report_group')}</InlineFormLabel>
+        <div className={flexWrapping}>
+          <InlineFormLabel tooltip={intl.get('report_group_description')}>{intl.get('report_group')}</InlineFormLabel>
 
-        <Select
-          value={reportGroups
-            ?.filter((reportGroup: ReportGroup) => reportGroup.id === schedule.reportGroupID)
-            .map((reportGroup: ReportGroup) => ({
+          <Select
+            value={reportGroups
+              ?.filter((reportGroup: ReportGroup) => reportGroup.id === schedule.reportGroupID)
+              .map((reportGroup: ReportGroup) => ({
+                label: reportGroup.name,
+                description: reportGroup.description,
+                value: reportGroup,
+              }))}
+            options={reportGroups?.map((reportGroup: ReportGroup) => ({
               label: reportGroup.name,
               description: reportGroup.description,
               value: reportGroup,
             }))}
-          options={reportGroups?.map((reportGroup: ReportGroup) => ({
-            label: reportGroup.name,
-            description: reportGroup.description,
-            value: reportGroup,
-          }))}
-          onChange={(selected: SelectableValue<ReportGroup>) => {
-            onUpdate(ScheduleKey.REPORT_GROUP_ID, selected?.value?.id ?? '');
-          }}
-        />
+            onChange={(selected: SelectableValue<ReportGroup>) => {
+              onUpdate(ScheduleKey.REPORT_GROUP_ID, selected?.value?.id ?? '');
+            }}
+          />
+        </div>
+
+        {(schedule.interval || 0) > 2 ? (
+          <div className={flexWrapping}>
+            <InlineFormLabel tooltip={intl.get('report_day_description')}>{intl.get('report_day')}</InlineFormLabel>
+
+            <Input
+              css=""
+              defaultValue={1}
+              min={1}
+              name={intl.get('report_day')}
+              onChange={({ currentTarget: { value } }) => onUpdate(ScheduleKey.DAY_OF_INTERVAL, parseInt(value))}
```

Comment:
wouldn't let me build without `parseInt(value, 10)` : crying about missing radix param

Reply:
odd that mine didn't complain. I expected that it would 🤷‍♂️